api/server.go

```diff
@@ -25,15 +26,9 @@ import (
 
 var log = logging.Logger("api-server")
 
-type Config interface {
-	GetServerAddress() string
-	GetServerPort() int
-	GetNetworkString() string
-}
-
 // apiServer is an implementation of node.Server interface for serving HTTP based Centrifuge API
 type apiServer struct {
-	config   Config
+	config   config.Config
```

Comment:
Vimukthi and I discussed this earlier, with multi-tenant we will split up the current config and it will change enough that making some assumptions now might be a waste of time. We will tackle that later.

Reply:
We need to have a deep discussion on this config and construction of objects in the node. I feel we are not on the same page in terms of the final architecture we want to achieve.